src/framework/parsers/draco-worker.js

```diff
@@ -0,0 +1,259 @@
+
+        if (!status || !status.ok() || mesh.ptr === 0) {
+            result.error = 'Failed to decode draco asset';
+            return result;
+        }
+
+        // indices
+        const numIndices = mesh.num_faces() * 3;
+        const shortIndices = mesh.num_points() <= 65535;
+        const indicesSize = numIndices * (shortIndices ? 2 : 4);
+        const indicesPtr = draco._malloc(indicesSize);
+        if (shortIndices) {
+            decoder.GetTrianglesUInt16Array(mesh, indicesSize, indicesPtr);
+            result.indices = new Uint16Array(draco.HEAPU16.buffer, indicesPtr, numIndices).slice().buffer;
+        } else {
+            decoder.GetTrianglesUInt32Array(mesh, indicesSize, indicesPtr);
+            result.indices = new Uint32Array(draco.HEAPU32.buffer, indicesPtr, numIndices).slice().buffer;
+        }
+        draco._free(indicesPtr);
+
+        // vertices
+        const attributes = [];
+        for (let i = 0; i < mesh.num_attributes(); ++i) {
+            attributes.push(decoder.GetAttribute(mesh, i));
+        }
+
+        // order attributes
+        attributes.sort((a, b) => {
+            return (attributeOrder[a.attribute_type()] ?? attributeOrder.length) - (attributeOrder[b.attribute_type()] ?? attributeOrder.length);
+        });
+
+        // calculate total vertex size and attribute offsets
+        let totalVertexSize = 0;
+        const offsets = attributes.map((a) => {
+            const offset = totalVertexSize;
+            totalVertexSize += Math.ceil(attributeSizeInBytes(a) / 4) * 4;
+            return offset;
+        });
+
+        // we will generate normals if they're missing
+        const hasNormals = attributes.some(a => a.attribute_type() === 1);
+        const normalOffset = offsets[1];
+        if (!hasNormals) {
+            for (let i = 1; i < offsets.length; ++i) {
+                offsets[i] += 12;
+            }
+            totalVertexSize += 12;
+        }
+
+        // create vertex buffer
+        result.vertices = new ArrayBuffer(mesh.num_points() * totalVertexSize);
+
+        // decode and interleave the vertex data
+        const dst = new Uint8Array(result.vertices);
+        for (let i = 0; i < mesh.num_attributes(); ++i) {
+            const attribute = attributes[i];
+            const sizeInBytes = attributeSizeInBytes(attribute);
+            const ptrSize = mesh.num_points() * sizeInBytes;
+            const ptr = draco._malloc(ptrSize);
+            decoder.GetAttributeDataArrayForAllPoints(mesh, attribute, attribute.data_type(), ptrSize, ptr);
+            const src = new Uint8Array(draco.HEAPU8.buffer, ptr, ptrSize);
+
+            // pack
+            for (let j = 0; j < mesh.num_points(); ++j) {
+                for (let c = 0; c < sizeInBytes; ++c) {
+                    dst[j * totalVertexSize + offsets[i] + c] = src[j * sizeInBytes + c];
+                }
+            }
+
+            if (!hasNormals && attribute.attribute_type() === 0) {
```

Comment:
I'd vote for not using '0' but have that wrapped by a constant in this file somewhere.

Reply:
yeah nice, I'll do that and see what else could do with labelling.